# Library Initialization

In [1]:
import pandas as pd


# Dataset Exploration

In [3]:
df = pd.read_csv("data/theme_park_visitors.csv")
df.head(5)

,Visitor_ID,Age,Gender,Ticket_Type,Favorite_Attraction,Number_of_Rides,Spending_Amount
0,1,48.0,Male,Standard,Water Ride,4,77.0
1,2,16.0,Male,VIP,Bumper Cars,2,171.0
2,3,35.0,Male,Standard,Roller Coaster,4,79.0
3,4,40.0,Male,VIP,Ferris Wheel,12,186.0
4,5,52.0,Female,Standard,Haunted House,10,91.0


In [4]:
df.tail(5)

,Visitor_ID,Age,Gender,Ticket_Type,Favorite_Attraction,Number_of_Rides,Spending_Amount
98,99,58.0,Male,VIP,Roller Coaster,5,151.0
99,100,50.0,Female,Standard,Ferris Wheel,11,99.0
100,3,35.0,Male,Standard,Roller Coaster,4,79.0
101,11,31.0,Male,VIP,Roller Coaster,1,189.0
102,21,49.0,Female,Standard,Water Ride,3,83.0


In [5]:
df.shape

(103, 7)

In [6]:
df.columns

Index(['Visitor_ID', 'Age', 'Gender', 'Ticket_Type', 'Favorite_Attraction',
       'Number_of_Rides', 'Spending_Amount'],
      dtype='str')

In [8]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 103 entries, 0 to 102
Data columns (total 7 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Visitor_ID           103 non-null    int64  
 1   Age                  102 non-null    float64
 2   Gender               103 non-null    str    
 3   Ticket_Type          102 non-null    str    
 4   Favorite_Attraction  102 non-null    str    
 5   Number_of_Rides      103 non-null    int64  
 6   Spending_Amount      102 non-null    float64
dtypes: float64(2), int64(2), str(3)
memory usage: 5.8 KB


no.rows = 103, no.columns = 7

- The columns "Visitor_ID", "Age", "Number_of_Rides" and "Spending_Amount" are numerical variables

- The columns "Gender", "Ticket_Type", and "Favorite_Attraction" are categorical

In [9]:
df.dtypes

Visitor_ID               int64
Age                    float64
Gender                     str
Ticket_Type                str
Favorite_Attraction        str
Number_of_Rides          int64
Spending_Amount        float64
dtype: object

# Missing Values Assessment

In [10]:
df.isnull().sum()

Visitor_ID             0
Age                    1
Gender                 0
Ticket_Type            1
Favorite_Attraction    1
Number_of_Rides        0
Spending_Amount        1
dtype: int64

The columns "Age", "Ticket_Type", "Favorite_Attraction", and "Spending_Amount" contain missing values, and they have the same number of missing values

# Duplicate Records Assessment

In [11]:
df.duplicated().sum()

np.int64(3)

The total number of duplicate records is 3

# Ticket Type Analysis

In [12]:
df["Ticket_Type"].value_counts()

Ticket_Type
VIP         41
Standard    34
Premium     27
Name: count, dtype: int64

The "VIP" ticket is the most common ticket type, while the "Premium" is the least

# Attraction Popularity Analysis

In [13]:
df["Favorite_Attraction"].value_counts()

Favorite_Attraction
Ferris Wheel      25
Water Ride        21
Roller Coaster    21
Bumper Cars       18
Haunted House     17
Name: count, dtype: int64

The most popular attraction is "Ferris Wheel"

Top 3:
- Ferris Wheel
- Water Ride
- Roller Coaster

# Summary Statistics

In [14]:
df.describe()

,Visitor_ID,Age,Number_of_Rides,Spending_Amount
count,103.000000,102.000000,103.000000,102.000000
mean,49.368932,35.450980,7.912621,130.529412
std,29.352486,15.459706,4.572095,67.631457
min,1.000000,8.000000,1.000000,22.000000
25%,23.500000,22.250000,4.000000,79.500000
50%,49.000000,36.500000,7.000000,123.500000
75%,74.500000,49.000000,12.000000,189.750000
max,100.000000,60.000000,15.000000,250.000000


- Average Visitor Age => 35.450980
- Average Spending Amount => 130.529412
- Highest Spending Amount => 250
- Lowest Spending Amount => 22
- Average Number of Rides => 7.912621

# Spending Analysis

In [15]:
df["Spending_Amount"].sum()

np.float64(13314.0)

In [16]:
df["Spending_Amount"].mean()

np.float64(130.52941176470588)

- Total spending generated by all visitors => 13314
- Average spending per visitor => 130.52941176470588

# Insight Generation

- no.visitors => 103
- There are missing values in the dataset
- There are duplicate records
- The most popular ticket type => VIP
- The most popular attraction => Ferris Wheel
- The average visitor age => 35.450980
- Average spending amount => 130.529412
- Data quality asessment is important before analysis, because it allows us to study the structure of the data and detecting the outliers, duplicates and missing values

# Analyst Report

- The data contains some missing values and duplicates, so it will need some cleaning
- There is a spread between the lowest spending (22) and the highest spending (250), so the mean isn't the best to represent the spending, we can represent it by the median
- The average visitor age is 35, so we must work to attract the other ages